devito/operator/operator.py

```diff
@@ -871,11 +872,13 @@ def apply(self, **kwargs):
         # Post-process runtime arguments
         self._postprocess_arguments(args, **kwargs)
 
-        # Output summary of performance achieved
+        # In case MPI is used restrict result logging to one rank only
+        with switch_log_level(comm=args.comm):
+            return self._emit_apply_profiling(args)
+
         return self._emit_apply_profiling(args)
```

Comment:
emitting twice ?

Reply:
ha! Actually not emmiting twice, but dead code never executed! dropping